# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/systemservices/android_video_capture/video_capture_android.h
# Commit: 55e2a4c5cab9
# Full Hash: 55e2a4c5cab961f0fa8627a490c4d714584b3065
# Author: Dan Minor <dminor@mozilla.com>
# Date: 2020-02-11 21:47:36
# Regressor Bug: 1578073
# File Overlap Count: 2
# Description:
#   Bug 1578073 - Move android video capture code to dom/media/systemservices; r=jib
#   
#   Although originally part of webrtc.org, this code has subsequently been
#   removed by upstream. Moving it to under dom/media should make it clearer that
#   this is code that we are maintaining and simplify future upstream merges.
# ==============================================================================

diff -r 27b45ca46f24 -r 55e2a4c5cab9 dom/media/systemservices/android_video_capture/video_capture_android.h
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/dom/media/systemservices/android_video_capture/video_capture_android.h	Mon Feb 10 17:08:36 2020 +0000
@@ -0,0 +1,48 @@
+/*
+ *  Copyright (c) 2012 The WebRTC project authors. All Rights Reserved.
+ *
+ *  Use of this source code is governed by a BSD-style license
+ *  that can be found in the LICENSE file in the root of the source
+ *  tree. An additional intellectual property rights grant can be found
+ *  in the file PATENTS.  All contributing project authors may
+ *  be found in the AUTHORS file in the root of the source tree.
+ */
+
+#ifndef WEBRTC_MODULES_VIDEO_CAPTURE_MAIN_SOURCE_ANDROID_VIDEO_CAPTURE_ANDROID_H_
+#define WEBRTC_MODULES_VIDEO_CAPTURE_MAIN_SOURCE_ANDROID_VIDEO_CAPTURE_ANDROID_H_
+
+#include <jni.h>
+
+#include "modules/video_capture/android/device_info_android.h"
+#include "modules/video_capture/video_capture_impl.h"
+
+namespace webrtc {
+namespace videocapturemodule {
+
+class VideoCaptureAndroid : public VideoCaptureImpl {
+ public:
+  VideoCaptureAndroid();
+  virtual int32_t Init(const char* deviceUniqueIdUTF8);
+
+  virtual int32_t StartCapture(const VideoCaptureCapability& capability);
+  virtual int32_t StopCapture();
+  virtual bool CaptureStarted();
+  virtual int32_t CaptureSettings(VideoCaptureCapability& settings);
+  virtual int32_t SetCaptureRotation(VideoRotation rotation);
+
+  int32_t OnIncomingFrame(uint8_t* videoFrame, size_t videoFrameLength,
+                          int32_t degrees, int64_t captureTime = 0);
+
+ protected:
+  virtual ~VideoCaptureAndroid();
+
+  DeviceInfoAndroid _deviceInfo;
+  jobject _jCapturer;  // Global ref to Java VideoCaptureAndroid object.
+  VideoCaptureCapability _captureCapability;
+  VideoRotation _rotation;
+  bool _captureStarted;
+};
+
+}  // namespace videocapturemodule
+}  // namespace webrtc
+#endif  // WEBRTC_MODULES_VIDEO_CAPTURE_MAIN_SOURCE_ANDROID_VIDEO_CAPTURE_ANDROID_H_